components/lwp/lwp_pmutex.c

```diff
@@ -32,6 +33,25 @@ struct rt_pmutex
     rt_uint8_t type; /* pmutex type */
 };
 
+/*
+ * userspace mutex definitions in musl
+ */
+struct rt_umutex
+{
+    union
+    {
+        int __i[6];
+        volatile int __vi[6];
+        volatile void *volatile __p[6];
+    } __u;
+};
```

Comment:
```suggestion
typedef struct rt_umutex
{
    union
    {
        int __i[6];
        volatile int __vi[6];
        volatile void *volatile __p[6];
    } __u;
} *rt_umutex_t;
```
下文中使用rt_umutex_t代替那一大串结构体指针

Reply:
不建议。typedef容易搞晕，少用